### PR TITLE
QRcode generation

### PR DESCRIPTION
**Pull request by Light111989**

### CODE REVIEW

File: src/components/QRcode.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<template>
+<div>
+  <canvas id="canvas"></canvas>
+</div>
+</template>
+<script>
+import QRCode from 'qrcode'
+export default {
+  props: {
```
Comment (hemvanh) [inline src/components/QRcode.vue]: try to add in some more props for customize the QR code, like colors, sizes ,... ?

File: src/components/QRcode.vue [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<template>
+<div>
```
Comment (hemvanh) [inline src/components/QRcode.vue]: if there's only 1 canvas then maybe no need to wrap inside a <div>

File: src/components/QRcode.vue [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+<script>
+import QRCode from 'qrcode'
+export default {
+  props: {
+    qrcode: [String, Object],
+  },
+  mounted() {
+    var canvas = this.$el.querySelector('#canvas')
+    QRCode.toCanvas(canvas, this.qrcode, function(error) {
+      if (error) console.error(error)
```
Comment (hemvanh) [inline src/components/QRcode.vue]: use proper warning/error alert, dont use console.log
console.log is just for testing when in development

File: src/pages/Stores.vue [hunk 4]
```diff
@@ -1,9 +1,27 @@
 <template>
-  <div>Store</div>
+<div>
```
Comment (hemvanh) [inline src/pages/Stores.vue:2]: Stores.vue ->  no need this file ?

**Review by hemvanh — CHANGES_REQUESTED** [review]
pls adjust your code

File: src/components/QRcode.vue (hunk 2 above)
Comment (Light111989) [reply src/components/QRcode.vue]: I tried. but can't remove this

File: src/components/GenQRcode.vue [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    var canvas = this.$el.querySelector('#canvas')
+    QRCode.toCanvas(
+      canvas,
+      this.qrcode,
+      {
+        color: {
+          dark: this.color[0], 
+          light: this.color[1],
+        },
+        version: this.size,
```
Comment (hemvanh) [inline src/components/GenQRcode.vue]: version: this.size  , --> this means ? version is size ?

File: src/components/GenQRcode.vue [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      },
+      function(error) {
```
Comment (hemvanh) [inline src/components/GenQRcode.vue]: 1) try to use arrow function  like: err=>{...}
2) try to use _procError util

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/components/GenQRcode.vue (hunk 5 above)
Comment (Light111989) [reply src/components/GenQRcode.vue]: Ya. For version 1 they are 21x21, for version 2 25x25 e so on. Higher is the version, more are the storable data, and of course bigger will be the QR Code symbol.

File: src/pages/Stores.vue (hunk 4 above)
Comment (Light111989) [reply src/pages/Stores.vue:2]: ya no need

File: src/components/GenQRcode.vue [hunk 7]
```diff
@@ -24,14 +29,13 @@ export default {
...
+          dark: this.dark,
+          light: this.light,
         },
         version: this.size,
       },
-      function(error) {
-        if (error) alert(error)
-        alert('success!')
+      err => {
+        if (err != null) _procError(err)
```
Comment (hemvanh) [inline src/components/GenQRcode.vue]: no need to check for  : if (err != null)
coz err 100% not null, coz it's in the catch clause

**Review by hemvanh — CHANGES_REQUESTED** [review]

**Review by hemvanh — APPROVED** [review]

**Pull request merged**